tests/test_x509.py

```diff
@@ -173,6 +173,20 @@ def test_revoked_cert_retrieval(self, backend):
         # Check that len() works for CRLs.
         assert len(crl) == 12
 
+    def test_revoked_cert_retrieval_retain_only_revoked(self, backend):
+        """
+        This test attempts to trigger the crash condition described in
+        https://github.com/pyca/cryptography/issues/2557
+        PyPy does gc at its own pace, so it will only be reliable on CPython.
+        """
+        revoked = _load_cert(
+            os.path.join("x509", "custom", "crl_all_reasons.pem"),
+            x509.load_pem_x509_crl,
+            backend
+        )[11]
+        assert revoked.revocation_date == datetime.datetime(2015, 1, 1, 0, 0)
+        assert revoked.serial_number == 11
+
     def test_extensions(self, backend):
```

Comment:
I tested it locally in a repl, it fails.

On Wed, Dec 23, 2015 at 9:22 PM, Erik Trauschke <notifications@github.com>
wrote:

> In tests/test_x509.py
> <https://github.com/pyca/cryptography/pull/2558#discussion_r48392416>:
>
> > @@ -173,6 +173,20 @@ def test_revoked_cert_retrieval(self, backend):
> >          # Check that len() works for CRLs.
> >          assert len(crl) == 12
> >
> > +    def test_revoked_cert_retrieval_retain_only_revoked(self, backend):
> > +        """
> > +        This test attempts to trigger the crash condition described in
> > +        https://github.com/pyca/cryptography/issues/2557
> > +        PyPy does gc at its own pace, so it will only be reliable on CPython.
> > +        """
> > +        revoked = _load_cert(
> > +            os.path.join("x509", "custom", "crl_all_reasons.pem"),
> > +            x509.load_pem_x509_crl,
> > +            backend
> > +        )[11]
> > +        assert revoked.revocation_date == datetime.datetime(2015, 1, 1, 0, 0)
> > +        assert revoked.serial_number == 11
> > +
> >      def test_extensions(self, backend):
>
> Just curious, but does this test actually fail without the code changes?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/2558/files#r48392416>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084